src/gc/impl/conservative/gc.d

```diff
@@ -2080,7 +2080,8 @@ struct Gcx
         {
             // local stack is full, push it to the global stack
             assert(stackPos == stack.length);
-            toscan.push(ScanRange(p1, p2));
+            if (p1 + 1 < p2)
```

Comment:
> That would be better added in the above loops where we break without incrementing the pointer.

Incrementing could be done unconditionally, but I think having a simple common subexpression here might be better than writing back the value to p1.

> In fact you can just factor this out as common code between all branches.

That way you have to calculate `top` unconditionally. I guess it could be slightly better to also move the calculation of `base` into the conditional.

Reply:
> Could help a bit to better fit the normal loop body in the µop-cache.

`toscan.push()` increases the loop size quite a bit because `ScanRange.grow()` is inlined. Disabling that with `pragma(inline, false)` didn't have any impact on performance, though.